File: astroloop-frontend/src/api/cosmicCard.ts
import api from './client';

export interface CosmicCard {
  id: number;
  userName: string;
  date: string;
  cosmicMessage: string;
  careerInsight: string;
  loveInsight: string;
  moneyInsight: string;
  energyScore: number;
  luckyElement: string;
  zodiacSign: string;
  shareId: string;
  shareUrl: string;
  viewCount: number;
  shareCount: number;
  createdAt: string;
}

export const cosmicCardApi = {
  generate: () => api.post<CosmicCard>('/cosmic-card/generate'),
  getMyCards: () => api.get<CosmicCard[]>('/cosmic-card/my-cards'),
  getPublic: (shareId: string) => api.get<CosmicCard>(`/public/cosmic-card/${shareId}`),
  trackShare: (shareId: string) => api.post<CosmicCard>(`/cosmic-card/share/${shareId}`),
};

File: astroloop-backend/src/main/java/com/astroloop/entity/CosmicCard.java
package com.astroloop.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "cosmic_cards")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CosmicCard {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, unique = true)
    private String shareId;

    @Column(nullable = false)
    private String cosmicMessage;

    @Column(length = 500)
    private String careerInsight;

    @Column(length = 500)
    private String loveInsight;

    @Column(length = 500)
    private String moneyInsight;

    @Column(nullable = false)
    private Integer energyScore;

    @Enumerated(EnumType.STRING)
    private com.astroloop.enums.CosmicElement luckyElement;

    private String zodiacSign;

    @Column(nullable = false)
    private Integer viewCount = 0;

    @Column(nullable = false)
    private Integer shareCount = 0;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
